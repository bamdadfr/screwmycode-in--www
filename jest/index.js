export { JestWithStyledComponents } from './jest-with-styled-components'