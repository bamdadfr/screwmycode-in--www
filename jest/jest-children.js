import React from 'react'

/**
 * @function
 * @name JestChildren
 * @description jest: dummy children
 * @returns {React.ReactElement} - react component
 */
export function JestChildren () {

    return (
        <>
            <span>children</span>
            <span>children</span>
            <span>children</span>
        </>
    )

}