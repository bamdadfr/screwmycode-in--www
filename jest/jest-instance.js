import renderer from 'react-test-renderer'
import { JestWrapper } from './jest-wrapper'

export const JestInstance = (Component) => renderer.create (JestWrapper (Component)).toJSON ()