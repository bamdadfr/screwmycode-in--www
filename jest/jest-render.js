import React from 'react'
import { render } from '@testing-library/react'
import { JestWrapper } from './jest-wrapper'

/**
 * @function
 * @name JestRender
 * @description jest: get a render from testing-library
 * @param {React.ReactElement} Component - react component to render
 * @returns {object} - testing library render
 */
export function JestRender (Component) {

    return render (
        JestWrapper (Component),
    )

}