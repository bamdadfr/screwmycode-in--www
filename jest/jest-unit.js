import React from 'react'
import { shallow } from 'enzyme'
import { JestWithStyledComponents } from './jest-with-styled-components'

/**
 * @function
 * @name JestUnit
 * @description jest: get unit from enzyme
 * @param {React.ReactElement} Component - react component
 * @param {boolean} [wrap=false] - wrap with state and theme...
 * @returns {object} - enzyme shallow object
 */
export function JestUnit (Component, wrap = false) {

    if (wrap) {

        return shallow (
            JestWithStyledComponents (Component),
        )

    }

    return shallow (Component)

}