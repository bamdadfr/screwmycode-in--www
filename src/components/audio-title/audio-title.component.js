import React from 'react'
import { Title } from './audio-title.component.styles'

/**
 * @param {object} props react props
 * @param {string} props.title audio title
 * @returns {React.ReactElement} react component
 */
export function AudioTitleComponent ({ title }) {

    return (
        <>
            <Title>
                {title}
            </Title>
        </>
    )

}