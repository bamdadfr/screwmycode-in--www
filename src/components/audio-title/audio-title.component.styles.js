import styled from 'styled-components'

export const Title = styled.h2`
    margin-bottom: 1em;
`