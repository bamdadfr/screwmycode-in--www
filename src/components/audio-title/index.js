export { AudioTitleComponent } from './audio-title.component'