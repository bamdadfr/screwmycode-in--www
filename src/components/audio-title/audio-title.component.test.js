import React from 'react'
import { screen } from '@testing-library/react'
import { AudioTitleComponent } from './audio-title.component'
import { JestInstance, JestRender } from '../../../jest'

const setup = (title = 'title') => {

    const Component = () => <AudioTitleComponent title={title} />
    const instance = JestInstance (<Component/>)
    const render = JestRender (<Component/>)

    const elements = {
        'title': screen.getByRole ('heading'),
    }

    return {
        instance,
        render,
        elements,
    }

}

describe ('AudioTitleComponent', () => {

    describe ('unit testing', () => {

        describe ('instance', () => {

            it ('should be defined', () => {

                const { instance } = setup ()

                expect (instance).toBeDefined ()

            })

            it ('should match snapshot', () => {

                const { instance } = setup ()

                expect (instance).toMatchSnapshot ()
            
            })

        })

        describe ('styles', () => {

            it ('should have a defined classname', () => {

                const { instance } = setup ()

                expect (instance.props.className).toBeDefined ()

            })

            it ('should have a sc classname', () => {

                const { instance } = setup ()

                expect (instance.props.className).toContain ('sc')

            })

        })

        describe ('children', () => {

            it ('should have 1 child', () => {

                const { instance } = setup ()

                expect (instance.children).toHaveLength (1)

            })

        })

    })

    describe ('visual testing', () => {

        describe ('title', () => {

            it ('should be in the document', () => {

                const { title } = setup ().elements

                expect (title).toBeInTheDocument ()

            })

            it ('should be visible', () => {

                const { title } = setup ().elements

                expect (title).toBeVisible ()

            })

            it ('should be enabled', () => {

                const { title } = setup ().elements

                expect (title).toBeEnabled ()

            })

            it ('should have a title prop of test value', () => {

                const value = 'testing dummy placeholder value'
                const { elements } = setup (value)
                const { title } = elements

                expect (title).toHaveTextContent (value)

            })
        
        })
    
    })

})