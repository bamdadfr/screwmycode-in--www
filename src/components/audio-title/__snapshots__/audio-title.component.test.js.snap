// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioTitleComponent unit testing instance should match snapshot 1`] = `
<h2
  className="sc-195iept-0 Tqshg"
>
  title
</h2>
`;
